torch/_inductor/ir.py

```diff
@@ -2895,12 +2897,25 @@ def gen_kwarg(k, v):
 
     @classmethod
     def create(cls, kernel, *args, **kwargs):
-        (
-            example_output,
-            tensor_args,
-            non_tensor_args,
-            unflatten_args,
-        ) = cls.process_kernel(kernel, *args, **kwargs)
+        fake_incorrect_kernels = (
+            aten._fft_r2c.default,
```

Comment:
@eellison, may I know why there have such fake_incorrect_kernels? I meet an issue when running a  model with a dynamic shape(the model has ```aten._fft_c2c.default```).

Reply:
we need to fix it. cc @nkaretnikov who was working on updating these kernels. maybe some of these are fixed now ?

Reply:
@eellison I cannot say for sure right now. Would you prefer that I take a look at these?

@XiaobingSuper Do you have a repro or more details for issues with dynamic?

Reply:
FWIW I've also just remembered that there might be issues with the conj bit in fft. It's not modeled correctly on the Python side. I general fix is tricky, but maybe I can split out some of the changes from here:
https://github.com/pytorch/pytorch/pull/101835

Reply:
@XiaobingSuper would you provide a repro ?

Reply:
Yes, see the following case from HF GoogleFnet:

```
import torch
import torch._dynamo
torch._dynamo.config.dynamic_shapes = True
torch._dynamo.config.assume_static_by_default = False

@torch._dynamo.optimize("inductor", dynamic=True)
def fn(a):
    convert_element_type: c64[s0, 512, 768] = torch.ops.prims.convert_element_type.default(a, torch.complex64)
    _fft_c2c: c64[s0, 512, 768] = torch.ops.aten._fft_c2c.default(convert_element_type, [1, 2], 0, True)
    return _fft_c2c

inp = torch.randn(1, 512, 768)
torch._dynamo.mark_dynamic(inp, 0)
print(fn(inp)
```

there has an error:
```
  File "/home/xiaobing/pytorch-offical/torch/_inductor/graph.py", line 643, in run_node
    result = fallback_handler(n.target, add_to_fallback_set=False)(
  File "/home/xiaobing/pytorch-offical/torch/_inductor/lowering.py", line 1152, in handler
    return pytree.tree_map(
  File "/home/xiaobing/pytorch-offical/torch/_inductor/ir.py", line 3321, in create
    ) = cls.process_kernel(kernel, *args, **kwargs)
  File "/home/xiaobing/pytorch-offical/torch/_inductor/ir.py", line 2711, in process_kernel
    example_args.append(ir_node_to_tensor(x, guard_shape=True))
  File "/home/xiaobing/pytorch-offical/torch/_inductor/ir.py", line 200, in ir_node_to_tensor
    t = torch.empty_strided(
torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
RuntimeError: /home/xiaobing/pytorch-offical/build/aten/src/ATen/RegisterCPU.cpp:5164: SymIntArrayRef expected to contain only concrete integers
```

Reply:
@nkaretnikov @eellison, a repro test case is given, please help see it.

Reply:
Will take a look at this issue this week.

Reply:
Haven't looked too deep, but I do know that the repro passes for me if I remove `c2c` from `fake_incorrect_kernels`. So I've put up a draft PR removing all fft and svd ops to see if it breaks anything: https://github.com/pytorch/pytorch/pull/103616